code-challenges.test.js

```diff
@@ -15,37 +15,108 @@
+ 
+
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
+// Input: an object containing two numbers
+// Output: the two numbers combined.
+// Pseudo: Need to subtract the downvotes from the upvotes within the object and return the total.
+
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("rockTheVote", () => {
+  it("takes in an object that contains up votes and down votes and returns the end tally.", () => {
+    expect(rockTheVote(votes1)).toEqual(11)
+    expect(rockTheVote(votes2)).toEqual(-31)
+  })
+});
+
+// [error 1:   ● Test suite failed to run --- ReferenceError: Cannot access 'rockTheVote' before initialization]  ***dangit, I'll get there eventually. forgot to 'stringify' the function name in the test...***
+// [error 2: TypeError: rockTheVote(...).toEqual is not a function] ***not a function my butt...misallocated some parantheticals.***
+
 const votes1 = { upVotes: 13, downVotes: 2 }
 // Expected output: 11
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
 // b) Create the function that makes the test pass.
+const rockTheVote = (input) => {return (input.upVotes - input.downVotes)};  // ***I promise I'll stop.***
+
+// *****this seems too easy not to be a trap, and now I'm paranoid that I missed something... I guess the only tricky part was remembering that we're working with objects so you need to access it in a different manner from an array?*****
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
+// input: two arrays containing strings
+// output: one array with no duplicate values
+// Pseudo: Concatinate the two input arrays, filter out the duplicates, and return the new filtered array.
+
 // a) Create a test with an expect statement using the variables provided.
+describe("hotGlue", () => {
+  it("takes in two arrays as arguments and returns one array with no duplicate values", () => {
+    expect(hotGlueTwoOhCrapItsEverywhere(dataArray1, dataArray2)).toEqual(["array", "object", "number", "string", "Boolean", "null", "undefined"])
+  })
+})
 
+// [error PSYCHE!] WOOO first try!
 const dataArray1 = ["array", "object", "number", "string", "Boolean"]
 const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
 // b) Create the function that makes the test pass.
+const hotGlue = (inp1, inp2) => {despair = (inp1.concat(inp2)); let hope = new Set(despair); let triumph = Array.from(hope); return triumph;}      // *** I DON'T HAVE A PROBLEM YOU HAVE A PROBLEM I CAN STOP WHENEVER I WANT SHUTUP! ***
```

Comment:
With this one, if you are declaring those let variables, we want to drop them down for readability vs the one liner you had on question #1. But great logic!